plaidml/bridge/openvino/ops/roi_pooling.cpp

```diff
@@ -0,0 +1,136 @@
+  std::vector<T> indices(len);
+  for (int64_t i = 0; i < len; i++) {
+    indices[i] = static_cast<T>(start_point + stride * i);
+  }
+  return edsl::Constant(makeBuffer(dtype, {len}, indices), name);
+}
+
+std::tuple<edsl::Tensor, int, int> crop_resized(edsl::Tensor& I, std::vector<float>& coord, std::string method,
+                                                int64_t pooled_h, int64_t pooled_w) {
+  auto x_1 = coord[0];
+  auto y_1 = coord[1];
+  auto x_2 = coord[2];
+  auto y_2 = coord[3];
+
+  auto roi_width = std::max(x_2 - x_1, 1.0f);
+  auto roi_height = std::max(y_2 - y_1, 1.0f);
+
+  int kernel_h = static_cast<int>(std::floor(roi_height / pooled_h));
+  int kernel_w = static_cast<int>(std::floor(roi_width / pooled_w));
+
+  edsl::Tensor w_tensor, h_tensor;
+  edsl::InterpolationMode interpolation_mode;
+  if (method != "bilinear") {
+    h_tensor = indices_generator(static_cast<int>(x_1), pooled_h, kernel_h, DType::INT32, "indices_h");
+    w_tensor = indices_generator(static_cast<int>(y_1), pooled_w, kernel_w, DType::INT32, "indices_w");
+    interpolation_mode = edsl::InterpolationMode::NEAREST;
+  } else {
+    h_tensor = indices_generator(x_1, pooled_h, kernel_h, DType::FLOAT32, "indices_h");
+    w_tensor = indices_generator(y_1, pooled_h, kernel_w, DType::FLOAT32, "indices_w");
+    interpolation_mode = edsl::InterpolationMode::LINEAR;
+  }
+
+  auto gather_w = edsl::gather(I, w_tensor).axis(3).interpolationMode(interpolation_mode);
+  auto gather_h = edsl::gather(gather_w, h_tensor).axis(2).interpolationMode(interpolation_mode);
+
+  return std::make_tuple(gather_h, kernel_h, kernel_w);
+}
+
+}  // namespace
+
+namespace PlaidMLPlugin {
+
+void registerROIPooling() {
+  registerOp("ROIPooling", [](const Context& ctx) {
+    auto* layer = ngraph::as_type<ngraph::opset4::ROIPooling>(ctx.layer);
+    IE_ASSERT(ctx.operands.size() == 2);
+    auto I = ctx.operands.at(0);
+    auto I_shape = I.compute_shape().sizes();
+
+    auto pooled_shape = layer->get_output_size();
+    auto pooled_height = static_cast<int64_t >(pooled_shape[0]);
+    auto pooled_width = static_cast<int64_t >(pooled_shape[1]);
+    auto spatial_ratio = layer->get_spatial_scale();
+    auto method = layer->get_method();
+
+    const static int BOX_ELEMENT_SIZE = 5;
+    auto coords_box = cast_constant_operand<float>(1, layer);
+    IE_ASSERT((coords_box.size() % BOX_ELEMENT_SIZE) == 0);
+
+    std::vector<edsl::Tensor> ROI_pools;
+    // 2D input tensor of shape [NUM_ROIS, 5] describing box
+    for (auto index = coords_box.begin(); index != coords_box.end(); index += BOX_ELEMENT_SIZE) {
+      // consisting of 5 element tuples: [batch_id, x_1, y_1, x_2, y_2]
+      auto batch_id = *index;
+      std::vector<float> coord(index + 1, index + BOX_ELEMENT_SIZE);
+      // translate ROI coordinates from their input spatial scale to the scale used when pooling
+      for (int i = 1; i < coord.size(); i++) {
+        coord[i] *= spatial_ratio;
```

Comment:
Can you use edsl::index to build a Tensor for coord and then pass to crop_resized as a Tensor？ You can build two Tensors it is easier to write code.

Reply:
ok, I'm trying it.

Reply:
we use coord to calculate roi_width ans roi_height, then use them to get kernel_h and kernel_w. those two variable would be used as pool op's parameter, and they have to be `int` type. so, if we use edsl::index to build Tensor for coord, it's hard to get `int kernel_h`  and `int kernel_w` in `crop_resized`.

Reply:
I see, the pool op is accepting int, we can keep it as it is.

Reply:
ok